### DISCUSSION TITLE
POST API call to create ticket won't pass the data from those fields to the ticket creation

### DISCUSSION THREAD

**Discussion by rodrigolfh** [django-helpdesk/django-helpdesk#1215] [Q&A] [answered]
Hi! I modified the models adding three new fields. I now can create tickets using those fields through the django-admin interface.

Then I added the logic to filter according to those fields in one of the ticket serializers, which now works with the GET api/tickets endpoint.

Now I want to use the ticket creation POST API endpoint while passing those custom params, but when the ticket is created those fields are set to null.  I have been looking for the model or function that handles those requests and fill the distinct fields into the new entry, with no avail. Would anybody tell me where is it??!

Thanks in advance.

**Comment by Benbb96** [comment, marked as answer]
I believe you need to add those custom fields inside the `TicketSerializer`'s `fields` and also adapt the `TicketForm` to accept these new fields and modify the `_create_ticket` function for the creation of the Ticket : https://github.com/django-helpdesk/django-helpdesk/blob/b02e8d416275b7006bbd584f9a2643ac8e051377/helpdesk/forms.py#L336).

**Reply by rodrigolfh** [reply]
I already added the fields to the TicketSerializer's Meta, but when adding self.cleaned_data['custom_field'] i get a KeyError because cleaned_data does not contain any of my custom fields. 

Also from where does it get the cleaned_data variable? I ctrl-F'd the entire forms.py script and cannot find the import or assignation.

Edit: It worked! I added the form fields at the beginning of the class and it worked. Thanks!
